lambdas/issue-credential/src/handler.ts

```diff
@@ -1,77 +1,129 @@
-        dynamoDBClient,
-        functionConfig.tableNames.personIdentityTable,
-        session.sessionId,
-        "expiryDate"
-      );
-      logger.info("Successfully retrieved the person identity record.");
-
-      const ninoUser = await retrieveNinoUser(
-        functionConfig.tableNames.ninoUserTable,
-        dynamoDBClient,
-        session.sessionId
-      );
-      logger.info("Successfully retrieved the nino user record.");
+      logger.info("Verifiable Credential Structure generated successfully.");
 
       return {
         statusCode: 200,
-        body: JSON.stringify({ failedAttemptCount, personIdentity, ninoUser }),
+        body: JSON.stringify(vcClaimSet),
       };
     } catch (error) {
       return handleErrorResponse(error, logger);
     }
   }
+  private getCiMappings(vcConfig: VcCheckConfig, failedAttempts: AttemptItem[]): CiMappings {
+    logger.info("Generating contraIndicator mapping inputs.");
+    return {
+      contraIndicationMapping: vcConfig.contraIndicator.errorMapping,
+      contraIndicatorReasonsMapping: vcConfig.contraIndicator.reasonsMapping,
+      hmrcErrors: failedAttempts.map((item) => item.text),
+    } as CiMappings;
+  }
+
+  private async getCheckedUserData(accessToken: string) {
+    const sessionId = await retrieveSessionIdByAccessToken(
+      functionConfig.tableNames.sessionTable,
+      dynamoDBClient,
+      accessToken
+    );
+    logger.info("Successfully retrieved the session id.");
+
+    const session: SessionItem = await getSessionBySessionId(functionConfig.tableNames.sessionTable, sessionId);
+
+    logger.appendKeys({
+      govuk_signin_journey_id: session.clientSessionId,
+    });
+    logger.info("Successfully retrieved the session record.");
+
+    const failedAttempts = await getAttempts(
+      functionConfig.tableNames.attemptTable,
+      dynamoDBClient,
+      session.sessionId,
+      "FAIL"
+    );
+    logger.info(`Identified ${failedAttempts.count} failed attempts.`);
+
+    const personIdentity: PersonIdentityItem = await getRecordBySessionId(
+      dynamoDBClient,
+      functionConfig.tableNames.personIdentityTable,
+      session.sessionId,
+      "expiryDate"
+    );
+    logger.info("Successfully retrieved the person identity record.");
+
+    const ninoUser: NinoUser = await retrieveNinoUser(
+      functionConfig.tableNames.ninoUserTable,
+      dynamoDBClient,
+      session.sessionId
+    );
+    logger.info("Successfully retrieved the nino user record.");
+
+    return { failedAttempts, session, personIdentity, ninoUser };
+  }
+
+  private async generateJwtClaims(subject: string): Promise<JwtClass> {
+    return {
+      iss: functionConfig.audit.issuer,
+      jti: `urn:uuid:${randomUUID().toString()}`,
+      nbf: toEpochSecondsFromNow(),
+      exp: toEpochSecondsFromNow(
+        functionConfig.credentialIssuerEnv.maxJwtTtl,
+        functionConfig.credentialIssuerEnv.jwtTtlUnit as TimeUnits
+      ),
+      sub: subject,
+    };
+  }
```

Comment:
Is there a particular reason you chose to use private class functions here? What decides whether you'd use a class function or create a separate directory?

Only noticed this as it's slightly different to the way Mike and I split the code for the other functions, but I'm not saying it's a problem or blocker for release. If there's a really good reason it could be useful to consider using in the other guys too 👍